src/main/java/pro/java/hw16/PetrolStation.java

```diff
@@ -0,0 +1,52 @@
+package pro.java.hw16;
+
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class PetrolStation {
+
+    private static final String RESET = "\u001B[0m";
+    private static final String RED = "\u001B[31m";
+    private static final String GREEN = "\u001B[32m";
+    private static final String CYAN = "\u001B[36m";
+    private static final String BLUE = "\033[0;34m";
+    private final AtomicInteger stationFuelAmount;
+
+    public PetrolStation(AtomicInteger initialAmount) {
+        this.stationFuelAmount = initialAmount;
+    }
+
+    public synchronized void doRefuel(int fuelAmount) {
+        try {
+            if (fuelAmount > stationFuelAmount.get()) {
+                throw new IllegalArgumentException();
+            }
+
+            int refuelTime = getRandomRefuelTime();
+            sleep(refuelTime);
+
+            decreaseAmount(fuelAmount);
+            System.out.println(GREEN + "Refueled " + fuelAmount + " litres of fuel, remaining fuel in station: " + RESET
+                    + BLUE + stationFuelAmount.get() + RESET);
+        } catch (IllegalArgumentException e) {
+            System.out.println(CYAN + Thread.currentThread().getName() + RESET + ": "
+                    + RED + "Not enough fuel at the station to refuel!" + RESET);
+        }
+    }
+
+    private int getRandomRefuelTime() {
+        return ThreadLocalRandom.current().nextInt(3, 11) * 1000;
+    }
+
+    private void sleep(int milliseconds) {
+        try {
+            Thread.sleep(milliseconds);
+        } catch (InterruptedException e) {
+            System.err.println("Sleep interrupted: " + e.getMessage());
+        }
+    }
+
+    private void decreaseAmount(int fuelAmount) {
+        stationFuelAmount.getAndAdd(-fuelAmount);
```

Comment:
Please, use the method decrementAndGet()

Reply:
I don't think the **decrementAndGet()** method is useful here, because it can't get the value of **fuelAmount** as a delta, and as a result it only **returns -1** at each method call. So it is necessary to use a loop until the **fuelAmount** value ends and at each iteration just subtract -1 from fuelAmount. This is very bad for performance and memory usage!

Reply:
Or you can use another method 
amount.updateAndGet(amount -> amount - fuelAmount);
It will calculate amount in proper way

Reply:
Ok, rewritten to this variant.

But, from the point of view of simplicity and readability, my variant of the method looks better, because it directly indicates the intention to reduce the value by the value of fuelAmount.
The lambda variant may be useful for some specific or complex operations that need to be performed when updating, but in my case, it's just one subtraction operation.

Reply:
Well, yes and no) I see you use a synchronized method so it a lock whole method and doesn't matter how you deal with your logic inside. But if you don't use synchronized you need to handle your operations as atomic just to make it thread-safe in concurrent environment. The simpliest way for sure is to use synchronized method but it lock all logic inside and pretty highweight approach. To more flexable and low porfomance is do not use synchronized around the method but handle partly locking or atomic operations

Reply:
@kartius OK, merging the current branch to the master requires your approval